Add README to BLEmulator (#284)
Polidea/FlutterBleLib: lib/blemulator/README.md

  }

  @override
  Future<bool> onConnectRequest() async {
    await Future.delayed(Duration(milliseconds: 200));
    return super.onConnectRequest();
  }
}
```

This creates a peripheral that advertises every 800 milliseconds while peripheral scan is on,
 waits for 200 milliseconds after receiving connection request before agreeing to connect and has a single service 
 with three characteristics according to [SensorTag documentation](http://processors.wiki.ti.com/images/a/a8/BLE_SensorTag_GATT_Server.pdf).

The _convenienceName_ fields are optional and not used by the blemulator itself, but allow you to name created objects for better maintainability.

#### Changing advertisement data or scan info

For your convenience you can mark SimulatedService as advertised, but be aware that this doesn't validate the size of advertised services array inside advertisement data.

If you want to have more granular control over what is advertised, simply modify the _scanInfo_ property of the SimulatedPeripheral.
 The _scanInfo_ is broadcasted automatically while peripheral scan is on each _advertisementInterval_.
 
ScanInfo supports following data:
```dart
class ScanInfo {
  int rssi;

>   int mtu;

Reviewer: I think it is a mistake by putting `mtu` in `ScanInfo` — MTU is a property of a connection made to peripheral, not scan's property.

Author: That's true, it's now removed completely from there.

  bool isConnectable;
  int txPowerLevel;

  Uint8List manufacturerData;
  Map<String, Uint8List> serviceData;
  List<String> serviceUuids;

  String localName;
  List<String> solicitedServiceUuids;
  List<String> overflowUuids;

  ScanInfo(
      {this.rssi = defaultRssi,
      this.mtu = defaultMtu,
      this.isConnectable = true,
      this.txPowerLevel,
      this.manufacturerData,
      this.serviceData,
      this.serviceUuids,
      this.localName,
      this.solicitedServiceUuids,
      this.overflowUuids});
}
```
You can also provide it in the constructor of the SimulatedPeripheral.

#### Custom characteristic behaviour

> Blemulator does most of the heavy lifting for you adn takes care of the basic stuff, but there's always more complicated logic.

Reviewer: ```suggestion
Blemulator does most of the heavy lifting for you and takes care of the basic stuff, but there's always more complicated logic.
```

> If you need to validate values or a write to one characteristic has to trigger a change in the behaviour of another, you may need to extend SimulatedService or SimulatedCharacteristic classes.

Reviewer: ```suggestion
If you need to validate values or write to one characteristic that has to trigger a change in behaviour of a different characteristic, you may need to extend SimulatedService or SimulatedCharacteristic classes.
```

                    SimulatedCharacteristic(
                      uuid: _temperatureDataUuid,
                      value: Uint8List.fromList([0, 0, 0, 0]),
                      convenienceName: "IR Temperature Data",
                      isWritableWithoutResponse: false,
                      isWritableWithResponse: false,
                    ),
                    SimulatedCharacteristic(
                        uuid: _temperatureConfigUuid,
                        value: Uint8List.fromList([0]),
                        convenienceName: "IR Temperature Config"),
                    SimulatedCharacteristic(
                        uuid: _temperaturePeriodUuid,
                        value: Uint8List.fromList([50]),
                        convenienceName: "IR Temperature Period"),
                  ],
                  convenienceName: "Temperature service"),
            ]) {
    scanInfo.localName = localName;

    getCharacteristicForService(_serviceUuid, _temperatureConfigUuid)
        .monitor()
        .listen((value) {
      int valueAsInt = value.buffer.asByteData().getUint8(0);
      _readingTemperature = valueAsInt == 1 ? true : false;
    });

>     _emitTemperature();

Reviewer: Will this practically launch a new thread that will emit values?

Author: In a way. While this is the same thread, it runs like a saga in RN. Since the `_emitTemperature()` function is marked as async and there's no `await` keyword before this call, it won't pause the execution of this method.

There are isolates in Dart that are physical threads, but they cannot share any context with the main thread.
